horovod/runner/elastic/discovery.py

```diff
@@ -15,20 +15,50 @@
 
 import io
 import logging
+import random
 import threading
-
+import time
 from collections import defaultdict
 
 from horovod.runner.common.util import safe_shell_exec
 from horovod.runner.elastic.worker import HostUpdateResult
 
+# The default lower bound for cooldown period. If a range is provided,
+# the provided lower limit must be at or above this lower bound
+DEFAULT_COOLDOWN_LOWER_LIMIT_SECONDS = 1
+# The default upper bound for cooldown period. If a range is provided,
+# the provided upper limit must be at or below this upper bound
+DEFAULT_COOLDOWN_UPPER_LIMIT_SECONDS = 1 * 60 * 60
 
 class HostState(object):
-    def __init__(self):
+
+    def __init__(self, cooldown_range=None):
         self._event = threading.Event()
 
-        # TODO(travis): blacklisted hosts should have a timeout period that increases with each failure
         self._blacklisted = False
+        self._blacklist_count = 0
+        if cooldown_range:
+            HostState._validate_cooldown_range(cooldown_range)
+            self._cooldown_lower_limit = cooldown_range[0]
+            self._cooldown_upper_limit = cooldown_range[1]
+        else:
+            self._cooldown_lower_limit = -1
+            self._cooldown_upper_limit = -1
+        self._cooldown_period_end_ts = 0
+
+    @staticmethod
+    def _validate_cooldown_range(cooldown_range):
+        cooldown_lower_limit = cooldown_range[0]
+        cooldown_upper_limit = cooldown_range[1]
```

Comment:
Now that `cooldown_range` is a tuple, why not:

```
        cooldown_lower_limit, cooldown_upper_limit = cooldown_range
```

Here and in a few similar other places.

Reply:
Done